### ISSUE TITLE
Trying to find go executable in GOPATH

### ISSUE THREAD

**Issue by HankelBao** [fatih/vim-go#2257]
### What did you do? (required. The issue will be **closed** when not provided.)

I tried to install the plugin using vim-plug.

### What did you expect to happen?

After GoInstallBinaries, the required go executables are installed.

### What happened instead?

No executables are installed with error:

> vim-go: could not find 'go'. Run :GoInstallBinaries to fix it.

### Configuration (**MUST** fill this out):

* vim-go version:
current Master repo

* `vimrc` you used to reproduce (use a *minimal* vimrc with other plugins disabled; do not link to a 2,000 line vimrc):
```
call plug#begin("/path/to/my/vimplugdir")
Plug 'fatih/vim-go', { 'do': ':GoUpdateBinaries' }
call plug#end
```

* Vim version (first three lines from `:version`):
> NVIM v0.3.4
> Build type: RelWithDebInfo
> LuaJIT 2.0.5

* Go version (`go version`):
go version go1.12.4 windows/amd64

* Go environment (`go env`):
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\hankelbao\AppData\Local\go-build
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOOS=windows
set GOPATH=C:\Users\hankelbao\Projects\go
set GOPROXY=
set GORACE=
set GOROOT=C:\Users\hankelbao\scoop\apps\go\current
set GOTMPDIR=
set GOTOOLDIR=C:\Users\hankelbao\scoop\apps\go\current\pkg\tool\windows_amd64
set GCCGO=gccgo
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\HANKEL~1\AppData\Local\Temp\go-build850241626=/tmp/go-build -gno-record-gcc-switches

EDITED: 

go.exe is in my path and I can install/get go projects in powershell.

**Comment by bhcleek** [comment]
What does `:echo $PATH` show from within vim?

**Comment by HankelBao** [comment]
> C:\Program Files (x86)\Common Files\Intel\Shared Libraries\redist\intel64\compiler;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Users\hankelbao\scoop\apps\rustup\current\.cargo\bin;C:\Users\hanke
lbao\scoop\apps\llvm\current\bin;C:\Users\hankelbao\scoop\apps\yarn\current\Yarn\bin;C:\Users\hankelbao\scoop\apps\yarn\current\global\node_modules\.bin;C:\Users\hankelbao\scoop\apps\python\current\scripts;C:\Users\hankelbao\AppData\Local\Programs\Python\Launcher\;C:\Users\hankelba
o\Projects\go\bin;C:\Users\hankelbao\scoop\apps\gcc\current\bin;C:\Users\hankelbao\scoop\apps\nodejs\current\bin;C:\Users\hankelbao\scoop\apps\nodejs\current;C:\Users\hankelbao\scoop\shims;C:\Users\hankelbao\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Bandizip\;C:\Program
Files (x86)\llvm-3.4-tools-windows;C:\Users\hankelbao\AppData\Roaming\Python\Python37\Scripts;C:\Users\hankelbao\scoop\apps\go\current\bin;;C:\Users\hankelbao\scoop\apps\neovim\current\Neovim\bin

*C:\Users\hankelbao\scoop\apps\go\current\bin* is where go.exe could be found.

**Comment by bhcleek** [comment]
Does `:!go env` work from within Vim?

**Comment by HankelBao** [comment]
Sorry for the delay in reply.

Yes, it works, with the exact same result in PowerShell executing `go env`.

**Comment by bhcleek** [comment]
I'm not sure what's causing this for you. 

Does `:echo $PATH` from within vim show the complete value?

After that, can you try moving `C:\Users\hankelbao\scoop\apps\go\current\bin` to the beginning of PATH? And trying `GoInstallBinaries` again?

**Comment by HankelBao** [comment]
Yes, I tried them all and somehow the problem persists.

It is really frustrating because everything really seems to be fine. One thing to notice is that I actually have some tools installed like gocode or gogetdoc but the plugin requires me to download it again.

**Comment by bhcleek** [comment]
Did `:echo $PATH` from within vim show the complete PATH that you expect to see? Specifically, does it show `C:\Users\hankelbao\scoop\apps\go\current\bin`?

**Comment by HankelBao** [comment]
Yes.

**Comment by HankelBao** [comment]
I tried to set the GOPATH in env the same as GOROOT and the error `vim-go: could not find 'go'. Run :GoInstallBinaries to fix it.` disappears. However, it was unable to download or install anything.

I checked the source code of the plugin a little bit and that section seems to be checking whether the required plugins are installed in the system, right? Why does it report that my go executable is not found?
